File: backend/app/schemas/scheduler.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class SchedulerJob(BaseModel):
    jobId: str
    taskName: str
    name: str
    description: str
    scheduleType: str
    scheduleValue: str
    isPaused: bool
    lastStatus: str | None = None
    lastMessage: str | None = None
    lastError: str | None = None
    lastRunStartedAt: datetime | None = None
    lastRunFinishedAt: datetime | None = None
    lastDurationMs: int | None = None
    nextRunAt: datetime | None = None


class SchedulerRun(BaseModel):
    id: int
    jobId: str
    trigger: str
    status: str
    message: str | None = None
    error: str | None = None
    startedAt: datetime
    finishedAt: datetime | None = None
    durationMs: int | None = None


class SchedulerJobsResponse(BaseModel):
    jobs: list[SchedulerJob]


class SchedulerRunsResponse(BaseModel):
    runs: list[SchedulerRun]


class SchedulerDispatchResponse(BaseModel):
    jobId: str
    taskId: str | None = None
    dispatchedAt: datetime


class SchedulerHealthResponse(BaseModel):
    brokerOk: bool
    brokerError: str | None = None
    totalJobs: int
    pausedJobs: int
    latestFinishedAt: datetime | None = None
    timestamp: datetime

File: frontend/src/types/scheduler.ts
export type SchedulerJob = {
  jobId: string;
  taskName: string;
  name: string;
  description: string;
  scheduleType: string;
  scheduleValue: string;
  isPaused: boolean;
  lastStatus?: string | null;
  lastMessage?: string | null;
  lastError?: string | null;
  lastRunStartedAt?: string | null;
  lastRunFinishedAt?: string | null;
  lastDurationMs?: number | null;
  nextRunAt?: string | null;
};

export type SchedulerRun = {
  id: number;
  jobId: string;
  trigger: string;
  status: string;
  message?: string | null;
  error?: string | null;
  startedAt: string;
  finishedAt?: string | null;
  durationMs?: number | null;
};

export type SchedulerJobsPayload = {
  jobs: SchedulerJob[];
};

export type SchedulerRunsPayload = {
  runs: SchedulerRun[];
};

export type SchedulerDispatchPayload = {
  jobId: string;
  taskId?: string | null;
  dispatchedAt: string;
};

export type SchedulerHealthPayload = {
  brokerOk: boolean;
  brokerError?: string | null;
  totalJobs: number;
  pausedJobs: number;
  latestFinishedAt?: string | null;
  timestamp: string;
};
